gpu_bdb/queries/q04/gpu_bdb_query_04_dask_sql.py

```diff
@@ -0,0 +1,176 @@
+#
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+# Copyright (c) 2019-2020, BlazingSQL, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+import os
+
+from bdb_tools.cluster_startup import attach_to_cluster
+from bdb_tools.sessionization import get_sessions
+
+from bdb_tools.utils import (
+    benchmark,
+    gpubdb_argparser,
+    run_query,
+)
+
+from bdb_tools.readers import build_reader
+
+from dask.distributed import wait
+
+
+def abandonedShoppingCarts(df, DYNAMIC_CAT_CODE, ORDER_CAT_CODE):
```

Comment:
I wonder if we should just have these functions in a file and import across both sql and dataframe versions.